flume-ng-node/src/main/java/org/apache/flume/node/Application.java

```diff
@@ -83,16 +84,25 @@ public synchronized void start() {
   }
 
   @Subscribe
-  public synchronized void handleConfigurationEvent(MaterializedConfiguration conf) {
-    stopAllComponents();
-    startAllComponents(conf);
+  public void handleConfigurationEvent(MaterializedConfiguration conf) {
+    if (stopping) {
+      logger.info("Will not handle Configuration Event while stopping");
+      return;
+    }
+    synchronized (this) {
+      stopAllComponents();
+      startAllComponents(conf);
+      
+    }
   }
 
   public synchronized void stop() {
+    stopping = true;
     supervisor.stop();
     if (monitorServer != null) {
       monitorServer.stop();
     }
+    stopping = false;
```

Comment:
Thanks for your remark. I'm afraid I will have to rework my change a bit, because this single boolean flag will not entirely prevent the issue.